### ISSUE TITLE
Remove start & end from range options

### ISSUE THREAD

**Issue by vweevers** [Level/community#86]
**I propose that we remove the `start` and `end` options from iterators & streams. Without doing a soft deprecation, because they were documented as legacy options [5 years ago](https://github.com/Level/leveldown/commit/e37fa20f219eeebf435d4a5904c54a17972419ce). I prefer ripping off the band aid and throwing an error if `start` or `end` properties are present in `options`.**

@Level/core Anyone opposed? Give a thumbs up or down. Keep reading for the motivation.

---

The `start/end` options are more surprising in reverse mode. While `ltgte` options fulfill the documented promise of "the returned entries are the same, but in reverse", `start/end` options must be swapped by the user to achieve that.

The relevant code to handle the whole set of options - which I think in many implementations has been built up through trial & error, putting trust in the extensive tests - is hard to follow, and hard to change.

In addition, there are rules of precedence between `start/end` and `ltgte` although a formal decision was never made (https://github.com/Level/abstract-leveldown/issues/70) and tests don't cover it.

**Comment by juliangruber** [comment]
I would still favor a soft deprecation, so we don't surprise anyone, but if you want to go ahead and remove it right away that should work too I guess

**Comment by vweevers** [comment]
@juliangruber It'll be released as semver-major across the board, with the usual changelogs and upgrade guides. For the folks that missed both the 5-year-long documented notice and the upgrade guides, we'll throw an error. I think that's more than enough.

**Comment by ralphtheninja** [comment]
This will greatly simplify the leveldown C++ code.

**Comment by vweevers** [comment]
Work is tracked in https://github.com/Level/community/issues/98.
